sdk/keyvault/keyvault-keys/test/public/utils/testAuthentication.ts

```diff
@@ -19,7 +19,11 @@ const replaceableVariables = {
   AZURE_KEYVAULT_ATTESTATION_URI: "https://skr_attestation.azure.net/",
 };
 
-export async function authenticate(that: Context, version: string): Promise<any> {
+export async function authenticate(
+  that: Context,
+  version: string,
+  testClientMaker?: (client: KeyClient) => TestClientInterface
```

Comment:
This feels a little odd to me - what is this used for? Sorry, I forgot 😄

Reply:
it’s to allow for the replacement of the TestClient with the InternalTestClient. The TestClient at the moment has some private references.